#!/usr/bin/env bash
#
# Wraps pyenv install
# usage: . $0
#
if [[ ! $PYENV_ROOT ]]; then
    echo 'You must set PYENV_ROOT in bashrc before calling' 1>&2
    exit 1
fi
if [[ ! -d $PYENV_ROOT ]]; then
    # works with pyenv 2+
    bivio_path_insert "$PYENV_ROOT"/bin 1
    bivio_path_insert "$PYENV_ROOT"/shims 1
    curl -s -S -L https://pyenv.run | bash
    # Newer versions of patch do not like relative file names. Give this warning:
    # 'Ignoring potentially dangerous file name ../Python-2.7.8/Lib/site.py'
    # Updating the patches this way fixes the problem
    find "$PYENV_ROOT" -name \*.patch -print0 | xargs -0 -n 100 perl -pi -e 's{^(\+\+\+|--- |diff.* )\.\./}{$1}'
fi

# Reset the environment
source "$HOME"/.bashrc

PYTHON_CONFIGURE_OPTS="${BIVIO_PYENV_VALGRIND:+--without-pymalloc --with-pydebug --with-valgrind} --enable-shared" pyenv install "$_bivio_pyenv_global_version"
pyenv global "$_bivio_pyenv_global_version"
